libraries/Microsoft.Bot.Builder.Expressions/BuiltInFunctions.cs

```diff
@@ -1854,92 +2965,122 @@ private static (object, string) SubArray(Expression expression, object state)
+                                    (value, error) = UriHost(uri);
+                                }
+                                else
+                                {
+                                    error = $"{expr} can't evaluate.";
                                 }
                             }
                             else
                             {
-                                error = $"{expr} can't evaluate.";
+                                error = $"{expr} should have one parameters";
                             }
                         }
+
                         return (value, error);
                     },
                     ReturnType.String,
-                    (expr) => ValidateOrder(expr, new[] { ReturnType.String }, ReturnType.String, ReturnType.Number, ReturnType.String)),
+                    ValidateUnary),
                 new ExpressionEvaluator(
-                    ExpressionType.DateReadBack,
-                    ApplyWithError(
-                        args =>
+                    ExpressionType.UriPath,
+                    (expr, state) =>
+                    {
+                        object value = null;
+                        string error = null;
+                        IReadOnlyList<dynamic> args;
+                        (args, error) = EvaluateChildren(expr, state);
+                        if (error == null)
                         {
-                            object result = null;
-                            string error;
-                            (result, error) = ParseISOTimestamp((string) args[0]);
-                            if (error == null)
+                            if (args.Count == 1)
                             {
-                                var timestamp1 = (DateTime) result;
-                                (result, error) = ParseISOTimestamp((string) args[1]);
-                                if (error == null)
+                                if (args[0] is string uri )
                                 {
-                                    var timestamp2 = (DateTime) result;
-                                    var timex = new TimexProperty(timestamp2.ToString("yyyy-MM-dd"));
-                                    result = TimexRelativeConvert.ConvertTimexToStringRelative(timex, timestamp1);
+                                    (value, error) = UriPath(uri);
+                                }
+                                else
+                                {
+                                    error = $"{expr} can't evaluate.";
                                 }
                             }
-                            return (result, error);
-                        },
-                        VerifyString),
+                            else
+                            {
+                                error = $"{expr} should have one parameters";
```

Comment:
no need to throw param count error for all functions, because validate has done this work.